### PR TITLE
[#11206] Add redundant_interface argument to cloud router interface

### PR DESCRIPTION
**Pull request by LucaPrete**
Add optional redundant_interface argument to google_compute_router_interface resource.

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

```release-note:enhancement
compute: Added optional `redundant_interface` argument to `google_compute_router_interface` resource
```

### CODE REVIEW

File: mmv1/third_party/terraform/resources/resource_compute_router_interface.go.erb [hunk 1]
```diff
@@ -137,6 +145,22 @@ func resourceComputeRouterInterfaceCreate(d *schema.ResourceData, meta interface
 
 	iface := &compute.RouterInterface{Name: ifaceName}
 
+	if riVal, ok := d.GetOk("redundant_interface"); ok {
+		ifaceFound := false
```
Comment (rileykarson) [inline mmv1/third_party/terraform/resources/resource_compute_router_interface.go.erb:149]: I'm not sure I understand the purpose of this section, from L148 -> L159. Could it be redundant with the search for a duplicate interface in L139 -> L144?

Comment (LucaPrete) [reply mmv1/third_party/terraform/resources/resource_compute_router_interface.go.erb:149]: Hello @rileykarson !

I don't think it's redundant, but maybe I'm missing something.
What these lines do is to check that the name of the redundant interface given as argument is an existing interface of the router.

Comment (LucaPrete) [reply mmv1/third_party/terraform/resources/resource_compute_router_interface.go.erb:149]: Hello @rileykarson. Could you please help me to close this? I need to send another patch which will need to rebase on this one. Thanks!

Comment (rileykarson) [reply mmv1/third_party/terraform/resources/resource_compute_router_interface.go.erb:149]: Oh, I understand now, thanks! The variable naming mixed me up, this checks the _redundant_ interface exists.

I think it'd be preferable to omit the check here, actually, and pass it off to the API. Clientside validations in Terraform are valuable when they're done at plan time- effectively, when they're *not* part of the CRUD methods- or when they save significant amounts of time (at least tens of seconds). The API's (hopefully) gonna have a more complete, and accurate error message, and should return pretty quickly.

**Comment by LucaPrete** [comment]
@rileykarson done!

**Review by rileykarson — APPROVED** [review]

**Pull request merged**